packages/vite/src/node/plugins/worker.ts

```diff
@@ -42,61 +71,100 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
           code: `import '${ENV_PUBLIC_PATH}'\n` + _
         }
       }
-      if (
-        query == null ||
-        (query && (query.worker ?? query.sharedworker) == null)
-      ) {
+
+      const workerConstructor = buildWorkerConstructor(query)
+      if (query == null || workerConstructor == null) {
         return
       }
 
       let url: string
       if (isBuild) {
-        if (query.inline != null) {
-          // bundle the file as entry to support imports and inline as blob
-          // data url
-          const rollup = require('rollup') as typeof Rollup
-          const bundle = await rollup.rollup({
-            input: cleanUrl(id),
-            plugins: config.plugins as Plugin[]
+        // bundle an inline script version of the worker
+        const { rollup } = require('rollup') as typeof Rollup
+        const bundle = await rollup({
+          input: cleanUrl(id),
+          plugins: config.plugins as Plugin[]
+        })
+
+        let code: string
+        try {
+          const { output } = await bundle.generate({
+            format: 'es',
```

Comment:
For monaco-editor, it would be useful if this was iife.
From #2494:
> Firefox was running into trouble with default exports still being exported (something like `unexpected top level export`)

Reply:
Any reason this was resolved but left unchanged?

Reply:
Hmm, thought I already changed this. Will push again!